cmd/soci/commands/create.go

```diff
@@ -98,10 +104,11 @@ var CreateCommand = cli.Command{
 
 		for _, plat := range ps {
 			builder, err := soci.NewIndexBuilder(cs, blobStore,
-				soci.WithMinLayerSize(minLayerSize),
-				soci.WithSpanSize(spanSize),
-				soci.WithBuildToolIdentifier(buildToolIdentifier),
-				soci.WithPlatform(plat))
+				append(builderOpts,
+					soci.WithMinLayerSize(minLayerSize),
+					soci.WithSpanSize(spanSize),
+					soci.WithBuildToolIdentifier(buildToolIdentifier),
```

Comment:
Given these 3 and `WithLegacyRegistrySupport` are the same and only `plat` is different, should we just put them to `builderOpts` and only append `plat` here. E.g.,

in line 66-70 (also need to move it to after getting minLayerSize and spanSize):

```golang
		builderOpts := []soci.BuildOption{
           soci.WithMinLayerSize(minLayerSize),
		   soci.WithSpanSize(spanSize),
		   soci.WithBuildToolIdentifier(buildToolIdentifier),
         }
		if cliContext.Bool(internal.LegacyRegistryFlagName) {
			builderOpts = append(builderOpts, soci.WithLegacyRegistrySupport)
		}
```

in line 106:

```golang
builder, err := soci.NewIndexBuilder(cs, blobStore, append(builderOpts, soci.WithPlatform(plat)))
```